src/deploy/AllocatorInit.sol

```diff
@@ -0,0 +1,141 @@
+    // TODO: add to ilk registry?
+
+    // Should be called from Pause Proxy
+    function initAllocator(
+        DssInstance memory dss,
+        AllocatorSharedInstance memory sharedInstance,
+        AllocatorCoreInstance memory coreInstance,
+        AllocatorConfig memory config
+    ) internal {
+
+        bytes32 ilk = VaultLike(coreInstance.vault).ilk();
+        dss.vat.init(ilk);
+        dss.vat.slip(ilk, coreInstance.vault, int256(1_000_000 * WAD));
+        dss.jug.init(ilk);
+
+        require(config.debtCeiling < WAD, "AllocatorInit/incorrect-ilk-line-precision");
+        dss.vat.file(ilk, "line", config.debtCeiling * RAD);
+        dss.vat.file("Line", dss.vat.Line() + config.debtCeiling * RAD);
+
+        dss.spotter.file(ilk, "pip", sharedInstance.oracle);
+        dss.spotter.file(ilk, "mat", RAY); // TODO: do we need this?
+        dss.spotter.poke(ilk);
+
+        RolesLike(sharedInstance.roles).setIlkAdmin(ilk, config.allocatorProxy);
+        RegistryLike(sharedInstance.registry).file(ilk, "buffer", coreInstance.buffer);
+    }
+
+    // TODO: sanity checks
+    // TODO: make sure it is agreed that only core contracts (vault + buffer) are added to changelog
+
+    // // Should be called from Allocator Proxy
+    function initCore(
+        DssInstance memory dss,
+        AllocatorSharedInstance memory sharedInstance,
+        AllocatorCoreInstance memory coreInstance,
+        CoreConfig memory config
+    ) internal {
+        VaultLike(coreInstance.vault).init();
+        VaultLike(coreInstance.vault).file("jug", address(dss.jug));
+        BufferLike(coreInstance.buffer).approve(config.nst, coreInstance.vault, type(uint256).max);
+
+        bytes32 ilk = VaultLike(coreInstance.vault).ilk();
+        RolesLike(sharedInstance.roles).setUserRole(ilk, config.facilitator, config.operatorRole, true);
+        RolesLike(sharedInstance.roles).setRoleAction(ilk, config.operatorRole, coreInstance.vault, VaultLike.draw.selector, true);
+        RolesLike(sharedInstance.roles).setRoleAction(ilk, config.operatorRole, coreInstance.vault, VaultLike.wipe.selector, true);
+    }
+
+    // TODO: sanity checks
+    // TODO: not that we do not approve NST from the buffer, as it could be any token that we want to swap
+
+    // Should be called from Allocator Proxy
+    function initFunnels(
+        AllocatorSharedInstance memory sharedInstance,
+        AllocatorCoreInstance memory coreInstance,
+        AllocatorFunnelsInstance memory funnelsInstance,
+        FunnelsConfig memory config
+    ) internal {
+
+        bytes32 ilk = VaultLike(coreInstance.vault).ilk();
+        RolesLike(sharedInstance.roles).setRoleAction(ilk, config.operatorRole, funnelsInstance.swapper, SwapperLike.swap.selector, true);
+        // TODO: same for the depositor
+
+        // TODO: are we fine with the same role ("operatorRole") for different entities (facilitator, stableSwapper...) operating on different contracts (vault, swapper..)?
```

Comment:
I think for the automation we could create a different role

Reply:
done